zerver/tests/test_hotspots.py

```diff
@@ -41,6 +41,57 @@ def test_send_all(self) -> None:
         with self.settings(DEVELOPMENT=True, ALWAYS_SEND_ALL_HOTSPOTS = True):
             self.assertEqual(len(ALL_HOTSPOTS), len(get_next_hotspots(self.user)))
 
+    def test_three_users_with_same_email_all_done(self) -> None:
+        for hotspot in ALL_HOTSPOTS:
+            do_mark_hotspot_as_read(self.user, hotspot)
+
+        user_in_red = do_create_user(
+            self.user.delivery_email,
+            'password',
+            do_create_realm("red", "red"),
+            'user',
+        )
+
+        user_in_blue = do_create_user(
+            self.user.delivery_email,
+            'password',
+            do_create_realm("blue", "blue"),
+            'user',
+        )
+
+        self.assertEqual(get_next_hotspots(user_in_red), [])
+        self.assertEqual(get_next_hotspots(user_in_blue), [])
+
+    def test_three_users_with_same_email_some_done_some_not(self) -> None:
+        for hotspot in ["intro_reply", "intro_streams"]:
+            do_mark_hotspot_as_read(self.user, hotspot)
+
+        user_in_red = do_create_user(
+            self.user.delivery_email,
+            'password',
+            do_create_realm("red", "red"),
+            'user',
+        )
+
+        user_in_blue = do_create_user(
+            self.user.delivery_email,
+            'password',
+            do_create_realm("blue", "blue"),
+            'user',
+        )
+
+        hotspots = get_next_hotspots(self.user)
+        self.assertEqual(len(hotspots), 1)
+        self.assertEqual(hotspots[0]['name'], 'intro_gear')
+
+        hotspots = get_next_hotspots(user_in_red)
+        self.assertEqual(len(hotspots), 1)
+        self.assertEqual(hotspots[0]['name'], 'intro_gear')
+
+        hotspots = get_next_hotspots(user_in_blue)
+        self.assertEqual(len(hotspots), 1)
+        self.assertEqual(hotspots[0]['name'], 'intro_gear')
+
```

Comment:
I think we can also mark intro_gear as read after this and then check for get_next_hotspots for the all 3 users again?